### PR TITLE
Models

### PR DESCRIPTION
**Pull request by RomanukAlex**

### CODE REVIEW

File: repair-service/dao-api/pom.xml [hunk 1]
```diff
@@ -16,6 +16,11 @@
         <artifactId>model</artifactId>
         <version>0.0.1-SNAPSHOT</version>
     </dependency>
+
+    <dependency>
```
Comment (aleksieiev) [inline repair-service/dao-api/pom.xml]: Looks like it had to be in another PR

File: repair-service/model/src/main/java/com/epam/repair/model/Address.java [hunk 2]
```diff
@@ -0,0 +1,137 @@
...
+    @Column(name="address_id", updatable=false, nullable=false)
+    private Integer addressId;
+
+    @Column(name="address_number_house", length = 40, nullable=false)
+    private String houseNumber;
+
+    @Column(name="address_number_apartment", length = 40, nullable=false)
+    private String apartmentNumber;
+
+    @ManyToOne(fetch = FetchType.EAGER)
```
Comment (aleksieiev) [inline repair-service/model/src/main/java/com/epam/repair/model/Address.java:29]: Are you sure that `EAGER` should be here?

File: repair-service/model/src/main/java/com/epam/repair/model/Device.java [hunk 3]
```diff
@@ -0,0 +1,175 @@
+package com.epam.repair.model;
+
+import javax.persistence.GeneratedValue;
+import javax.persistence.GenerationType;
+import javax.persistence.Id;
+
+/**
+ * Class represents entity with device information of repairing service.
+ */
+public class Device {
```
Comment (aleksieiev) [inline repair-service/model/src/main/java/com/epam/repair/model/Device.java:12]: `@Entity` absent

File: repair-service/rest-app/pom.xml [hunk 4]
```diff
@@ -17,12 +17,19 @@
...
 
         <dependency>
             <groupId>org.springframework.boot</groupId>
             <artifactId>spring-boot-starter</artifactId>
         </dependency>
+
+        <dependency>
+            <groupId>org.springframework.boot</groupId>
+            <artifactId>spring-boot-starter-test</artifactId>
+        </dependency>
```
Comment (aleksieiev) [inline repair-service/rest-app/pom.xml]: it should be in another PR

File: repair-service/rest-app/src/main/java/com/epam/repair/rest_app/RestApplication.java [hunk 5]
```diff
@@ -0,0 +1,15 @@
+package com.epam.repair.rest_app;
```
Comment (aleksieiev) [inline repair-service/rest-app/src/main/java/com/epam/repair/rest_app/RestApplication.java]: bad name of the package `rest_app`
I personally haven't ever seen names with `_` in the name

File: repair-service/service-api/src/main/java/com/epam/repair/rest_app/AddressService.java [hunk 6]
```diff
@@ -0,0 +1,17 @@
...
+import com.epam.repair.model.Address;
+import java.util.List;
+
+/**
+ * Address Service Interface.
+ */
+public interface AddressService {
+
+    /**
+     * Find all list.
```
Comment (aleksieiev) [inline repair-service/service-api/src/main/java/com/epam/repair/rest_app/AddressService.java]: not sure it find list.
it find all addresses

File: repair-service/service-api/src/main/java/com/epam/repair/rest_app/CityService.java [hunk 7]
```diff
@@ -0,0 +1,17 @@
+package com.epam.repair.rest_app;
```
Comment (aleksieiev) [inline repair-service/service-api/src/main/java/com/epam/repair/rest_app/CityService.java]: btw
why service is in `rest_app` package?
thing it should be in a separate package

File: repair-service/service/src/main/java/com/epam/repair/rest_app/AddressServiceImpl.java [hunk 8]
```diff
@@ -0,0 +1,27 @@
...
+import org.springframework.transaction.annotation.Transactional;
+
+import java.util.List;
+
+/**
+ * Address Service Interface implementation.
+ */
+@Service
+@Transactional
+public class AddressServiceImpl implements AddressService{
```
Comment (aleksieiev) [inline repair-service/service/src/main/java/com/epam/repair/rest_app/AddressServiceImpl.java]: checkstyle should fails here.
guess why?

File: repair-service/service/src/main/java/com/epam/repair/rest_app/CityServiceImpl.java [hunk 9]
```diff
@@ -0,0 +1,27 @@
...
+import org.springframework.transaction.annotation.Transactional;
+
+import java.util.List;
+
+/**
+ * City Service Interface implementation.
+ */
+@Service
+@Transactional
+public class CityServiceImpl implements CityService{
```
Comment (aleksieiev) [inline repair-service/service/src/main/java/com/epam/repair/rest_app/CityServiceImpl.java]: checkstyle is crying

File: repair-service/service/src/main/java/com/epam/repair/rest_app/StreetServiceImpl.java [hunk 10]
```diff
@@ -0,0 +1,27 @@
...
+import org.springframework.transaction.annotation.Transactional;
+
+import java.util.List;
+
+/**
+ * City Service Interface implementation.
+ */
+@Service
+@Transactional
+public class StreetServiceImpl implements StreetService{
```
Comment (aleksieiev) [inline repair-service/service/src/main/java/com/epam/repair/rest_app/StreetServiceImpl.java]: checkstyle is dehydrated

**Review by aleksieiev — APPROVED** [review]

File: repair-service/dao-api/pom.xml (hunk 1 above)
Comment (RomanukAlex) [reply repair-service/dao-api/pom.xml]: Comment accepted. Test dependencies removed.

File: repair-service/model/src/main/java/com/epam/repair/model/Device.java (hunk 3 above)
Comment (RomanukAlex) [reply repair-service/model/src/main/java/com/epam/repair/model/Device.java:12]: Comment accepted. @Entity added.

File: repair-service/rest-app/pom.xml (hunk 4 above)
Comment (RomanukAlex) [reply repair-service/rest-app/pom.xml]: Comment accepted. Test dependencies removed.

File: repair-service/rest-app/src/main/java/com/epam/repair/rest_app/RestApplication.java (hunk 5 above)
Comment (RomanukAlex) [reply repair-service/rest-app/src/main/java/com/epam/repair/rest_app/RestApplication.java]: Comment accepted. Package name changed to rest-app.

File: repair-service/service-api/src/main/java/com/epam/repair/rest_app/AddressService.java (hunk 6 above)
Comment (RomanukAlex) [reply repair-service/service-api/src/main/java/com/epam/repair/rest_app/AddressService.java]: Comment accepted. Package name changed.

File: repair-service/service-api/src/main/java/com/epam/repair/rest_app/CityService.java (hunk 7 above)
Comment (RomanukAlex) [reply repair-service/service-api/src/main/java/com/epam/repair/rest_app/CityService.java]: Comment accepted. Package name changed.

File: repair-service/rest-app/src/main/java/com/epam/repair/rest_app/RestApplication.java (hunk 5 above)
Comment (aleksieiev) [reply repair-service/rest-app/src/main/java/com/epam/repair/rest_app/RestApplication.java]: `rest` enough

File: repair-service/service/src/main/java/com/epam/repair/rest_app/AddressServiceImpl.java (hunk 8 above)
Comment (RomanukAlex) [reply repair-service/service/src/main/java/com/epam/repair/rest_app/AddressServiceImpl.java]: Comment accepted. Package name changed.

File: repair-service/service/src/main/java/com/epam/repair/rest_app/CityServiceImpl.java (hunk 9 above)
Comment (RomanukAlex) [reply repair-service/service/src/main/java/com/epam/repair/rest_app/CityServiceImpl.java]: Comment accepted. Package name changed.

File: repair-service/service/src/main/java/com/epam/repair/rest_app/StreetServiceImpl.java (hunk 10 above)
Comment (RomanukAlex) [reply repair-service/service/src/main/java/com/epam/repair/rest_app/StreetServiceImpl.java]: Comment accepted. Package name changed.

File: repair-service/rest-app/src/main/java/com/epam/repair/rest_app/RestApplication.java (hunk 5 above)
Comment (RomanukAlex) [reply repair-service/rest-app/src/main/java/com/epam/repair/rest_app/RestApplication.java]: Comment accepted. Package name changed to rest.

File: repair-service/model/src/main/java/com/epam/repair/model/Address.java (hunk 2 above)
Comment (RomanukAlex) [reply repair-service/model/src/main/java/com/epam/repair/model/Address.java:29]: Why not? I thought, if I want to get all the address data completely, I need EAGER.

**Review by sergeyarhutik — APPROVED** [review]

**Comment by sergeyarhutik** [comment]
there are 2 approvals, but it's not available to merge:
"This branch cannot be rebased due to conflicts
Rebasing the commits of this branch on top of the base branch cannot be performed automatically due to conflicts encountered while reapplying the individual commits from the head branch. "
